hr_expense_invoice/models/hr_expense.py

```diff
@@ -10,17 +11,56 @@
 class HrExpense(models.Model):
     _inherit = "hr.expense"
 
+    sheet_id_state = fields.Selection(related="sheet_id.state", string="Sheet state")
     invoice_id = fields.Many2one(
         comodel_name="account.move",
         string="Vendor Bill",
         domain=[
             ("type", "=", "in_invoice"),
             ("state", "=", "posted"),
             ("invoice_payment_state", "=", "not_paid"),
+            ("expense_ids", "=", False),
         ],
         copy=False,
     )
 
+    def action_expense_create_invoice(self):
+        invoice_lines = [
+            (
+                0,
+                0,
+                {
+                    "product_id": self.product_id.id,
+                    "name": self.name,
+                    "price_unit": self.unit_amount,
+                    "quantity": 1,
```

Comment:
This change is not correct AFAIK

Reply:
These changes it's about data consistency .
If we change quantity=1 in invoice_id onchange, IMO it's correct to set exactly the same in other use cases.
What do you think about it?

Reply:
No, it isn't. Starting with, you are not multiplying the unit_amount * quantity in the price_unit, but the question is the consistency is needed from invoice to expense, not the reverse. Please leave it as it was.

Reply:
Sorry but if leave it as it was these comment https://github.com/OCA/hr-expense/pull/34#discussion_r615237666 will be pending right?

Reply:
This one still remains incorrect